src/mainwindow.cpp

```diff
@@ -1310,29 +1357,277 @@ void MainWindow::initializeSettingsDatabase()
+            m_netPurchaseDataReplyFirstAttempt->deleteLater();
+        }
+    });
+}
+
+/*!
+ * \brief MainWindow::verifyLicenseSignalsSlots
+ */
+void MainWindow::verifyLicenseSignalsSlots()
+{
+    connect(this, &MainWindow::tryPurchaseDataSecondAlternative, this, [this]() {
+        m_dataBuffer->clear();
+        m_netPurchaseDataReplySecondAttempt = m_netManager->get(m_reqAlt2);
+
+        connect(m_netPurchaseDataReplySecondAttempt, &QNetworkReply::readyRead, this,
+                [this]() { m_dataBuffer->append(m_netPurchaseDataReplySecondAttempt->readAll()); });
+        connect(m_netPurchaseDataReplySecondAttempt, &QNetworkReply::finished, this, [this]() {
+            if (m_netPurchaseDataReplySecondAttempt->error() != QNetworkReply::NoError) {
+                qDebug() << "Error : " << m_netPurchaseDataReplySecondAttempt->errorString();
+                qDebug() << "Failed second attempt at getting data. Using embedded data...";
+            } else {
+                qDebug() << "success 2";
+                m_paymentDetails = QJsonDocument::fromJson(*m_dataBuffer).object();
+            }
+            emit fetchingPaymentDetailsRemotelyFinished();
+            m_netPurchaseDataReplySecondAttempt->deleteLater();
+        });
+    });
+
+    connect(this, &MainWindow::fetchingPaymentDetailsRemotelyFinished, this, [this]() {
+        if (m_paymentDetails.isEmpty()) {
+            qDebug() << "Using default";
+
+            QJsonObject paymentDetailsDefault;
+            paymentDetailsDefault["purchase_pro_url"] = "https://www.get-notes.com/pricing";
+            paymentDetailsDefault["purchaseApiBase"] = "https://api.lemonsqueezy.com";
+            paymentDetailsDefault["activateLicenseEndpoint"] = "/v1/licenses/activate";
+            paymentDetailsDefault["validateLicenseEndpoint"] = "/v1/licenses/validate";
+            paymentDetailsDefault["deactivateLicenseEndpoint"] = "/v1/licenses/deactivate";
+            m_paymentDetails = paymentDetailsDefault;
+        }
+        //        else {
+        //            qDebug() << "Using fetched data:";
+        //            qDebug() << m_paymentDetails;
+        //        }
+        emit gettingPaymentDetailsFinished();
+    });
+
+    connect(this, &MainWindow::gettingPaymentDetailsFinished, this, [this]() {
+        QString validateLicenseEndpoint = m_paymentDetails["purchaseApiBase"].toString()
+                + m_paymentDetails["validateLicenseEndpoint"].toString();
+        QNetworkRequest request{ QUrl(validateLicenseEndpoint) };
+        QJsonObject licenseDataObject;
+        licenseDataObject["license_key"] = m_userLicenseKey;
+        QJsonDocument licenseDataDoc(licenseDataObject);
+        QByteArray postData = licenseDataDoc.toJson();
+        request.setHeader(QNetworkRequest::ContentTypeHeader, "application/json");
+        QNetworkReply *reply = m_netManager->post(request, postData);
+
+        connect(reply, &QNetworkReply::finished, this, [=]() {
```

Comment:
I also think this section looks particularly hard to read because of the awkward clang formatting.

Wouldn't it be possible to rewrite these lambda expressions as methods part of the class? At least I think that will improve readability a bit, because the indentation won't be pushed too far to the right.